src/main/java/org/jenkinsci/plugins/workflow/job/WorkflowRun.java

```diff
@@ -1024,7 +1024,15 @@ public int hashCode() {
     @Override public void doConsoleText(StaplerRequest req, StaplerResponse rsp) throws IOException {
         rsp.setContentType("text/plain;charset=UTF-8");
         try (OutputStream os = rsp.getCompressedOutputStream(req)) {
-            getLogText().writeLogTo(0, os);
+            // Similar to writeWholeLogTo but terminates even if !logText.complete:
+            long pos = 0;
+            while (true) {
+                long pos2 = getLogText().writeLogTo(pos, os);
```

Comment:
Interesting possibility: what if the log is being generated faster than it can be written to output?